neal/sampler.py

```diff
@@ -289,46 +360,78 @@ def _default_ising_beta_range(h, J):
-
-    if not abs_biases:
-        return [0.1, 1.0]
-
-    # Rough approximation of min change in energy when flipping a qubit
-    min_delta_energy = min(abs_biases)
-
-    # Combine absolute biases by variable
-    abs_bias_dict = defaultdict(int, {k: abs(v) for k, v in h.items()})
+    # Approximate worst and best cases of the [non-zero] energy signal (effective field)
+    # experienced per spin as function of neighbors: bias = h_i + sum_j Jij s_j:
+    sum_abs_bias_dict = defaultdict(int, {k: abs(v) for k, v in h.items()})
+    min_abs_bias_dict = {key: sum_abs_bias_dict[key] for key in sum_abs_bias_dict if sum_abs_bias_dict[key]!=0}
+    #This loop is slow, but is not a bottleneck for practical implementations of simulated annealing:
     for (k1, k2), v in J.items():
-        abs_bias_dict[k1] += abs(v)
-        abs_bias_dict[k2] += abs(v)
-
-    # Find max change in energy when flipping a single qubit
-    max_delta_energy = max(abs_bias_dict.values())
-
+        for k in [k1,k2]:
+            sum_abs_bias_dict[k] += abs(v)
+            if v != 0: #This is slow, but this routine is not a bottleneck:
+                if k in min_abs_bias_dict:
+                    min_abs_bias_dict[k] = min(abs(v),min_abs_bias_dict[k])
+                else:
+                    min_abs_bias_dict[k] = abs(v)
+                
     # Selecting betas based on probability of flipping a qubit
-    # Hot temp: We want to scale hot_beta so that for the most unlikely qubit flip, we get at least
-    # 50% chance of flipping.(This means all other qubits will have > 50% chance of flipping
-    # initially.) Most unlikely flip is when we go from a very low energy state to a high energy
-    # state, thus we calculate hot_beta based on max_delta_energy.
-    #   0.50 = exp(-hot_beta * max_delta_energy)
-    #
-    # Cold temp: Towards the end of the annealing schedule, we want to minimize the chance of
-    # flipping. Don't want to be stuck between small energy tweaks. Hence, set cold_beta so that
-    # at minimum energy change, the chance of flipping is set to 1%.
-    #   0.01 = exp(-cold_beta * min_delta_energy)
-    hot_beta = np.log(2) / max_delta_energy
-    cold_beta = np.log(100) / min_delta_energy
-
+    # Hot temp: We want to be able to quickly search the entire solution space (equilibrate).
+    # Scale hot_beta so that all spins are able to flip with probability at least 50%, which
+    # ensures mixing across all states is fast.
+    # The most unlikely flip is related to the largest energy gap that must be overcome, with
+    # Metropolis updates we require:
+    #   0.50 = exp(-hot_beta * max_delta_energy) - (1)
+    # This is solved as hot_beta = log(2)/max_delta_energy, max_delta energy is twice the
+    # effective field, we take a worst case of the effective field to be conservative.
+    # Max delta energy occurs when all biases are aligned, and contribute without frustration:
+    max_effective_field = max(sum_abs_bias_dict.values())
```

Comment:
`max()` fails on empty iterable arg (i.e. the whole function will fail on empty `h` and/or empty input BQM). Check out linux/mac tests.